content/200-concepts/100-components/02-prisma-client/250-opentelemetry-tracing.mdx

```diff
@@ -0,0 +1,324 @@
+    email: email,
+  },
+  include: {
+    posts: true,
+  },
+})
+```
+
+The trace is structured as follows:
+
+- `prisma:client`
+  - `prisma:engine`:
+    - `prisma:engine:connection`
+    - `prisma:engine:db_query`: details of the first SQL query or command...
+    - `prisma:engine:db_query`: ...details of the next SQL query or command...
+    - `prisma:engine:db_query`: ...and so on, for each SQL query or command
+
+## Considerations and prerequisites
+
+If your application sends a large number of spans to a [collector](https://opentelemetry.io/docs/collector/), this can have a significant performance impact. For information on how to minimize this impact, see [Reducing performance impact](#reducing-performance-impact).
+
+To use tracing, you must do the following:
+
+1. [Install the appropriate dependencies](#step-1-install-up-to-date-prisma-dependencies).
+1. [Enable the `tracing` feature flag in your Prisma schema file](#step-2-enable-the-feature-flag-in-your-prisma-schema-file).
+1. [Install OpenTelemetry packages](#step-3-install-opentelemetry-packages).
+
+## Get started with tracing in Prisma
+
+### Step 1. Install up-to-date Prisma dependencies
+
+Use version `4.2.0` or higher of the `prisma`, `@prisma/client`, and `@prisma/instrumentation` npm packages.
+
+```terminal
+npm install prisma@latest --save-dev
+npm install @prisma/client@latest --save
+npm install @prisma/instrumentation@latest --save
+```
+
+### Step 2: Enable the feature flag in your Prisma schema file
+
+In the `generator` block of your `schema.prisma` file, enable the `tracing` feature flag:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["tracing"]
+}
+```
+
+### Step 3: Install OpenTelemetry packages
+
+Finally, install the appropriate OpenTelemetry packages, as follows:
+
+```console
+npm install @opentelemetry/semantic-conventions @opentelemetry/exporter-trace-otlp-http @opentelemetry/instrumentation @opentelemetry/sdk-trace-base @opentelemetry/sdk-trace-node @opentelemetry/resources
+```
+
+### Register tracing in your application
+
+The following code provides a minimal tracing configuration for your application. This configuration should be tweaked for your application.
+
+```ts
+// Imports
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions'
+import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-http'
+import { registerInstrumentations } from '@opentelemetry/instrumentation'
+import { SimpleSpanProcessor } from '@opentelemetry/sdk-trace-base'
+import { NodeTracerProvider } from '@opentelemetry/sdk-trace-node'
+import { PrismaInstrumentation } from '@prisma/instrumentation'
+import { Resource } from '@opentelemetry/resources'
+
+// Configure the trace provider
+const provider = new NodeTracerProvider({
+  resource: new Resource({
+    [SemanticResourceAttributes.SERVICE_NAME]: 'example application',
+  }),
+})
+
+// Setup how spans are processed and exported
```

Comment:
Maybe it would be worth explaining that this is specifically for otlp compatible collectors (like jaeger)

Reply:
Good idea